### ISSUE TITLE
include_vars.py should have an optional 'namespace' parameter

### ISSUE THREAD

**Issue by bobobox** [ansible/ansible#9868]
##### Issue Type:

Feature Idea
##### Ansible Version:

1.8.1
##### Environment:

N/A
##### Summary:

The include_vars module is useful for bringing in JSON and YAML formatted data from a variety of sources, but in many cases the names in these sources are quite generic (ie. 'hosts', or 'name'). If the module accepted and added an (optional) 'namespace' parameter, this would prevent collisions and unexpected behavior for common variable/key names. For example, instead of 'hosts', it could be added to facts as 'my_external_data.hosts'.

In other words, the option would result in the variables being imported as the content of a dict named after the 'namespace' parameter. 
##### Steps To Reproduce:

Import some YAML or JSON formatted data using include_vars.
##### Expected Results:

Facts should be (optionally) imported into a custom namespace (ie. a dict named after the 'namespace' parameter) rather than directly into the "root" namespace.
##### Actual Results:

Facts are imported directly into the "root" namespace, making conflicts a likely possibility.

**Comment by jimi-c** [comment]
# Switch to Proposal

Hi!

As of April of 2016, we have started using the Ansible Proposal process for large feature ideas or changes in current functionality, such as this. Because this issue has been open for so long without any action, we are going to go ahead and close it at this time. If you are still interested in seeing this new feature get into Ansible, please submit a proposal for it using this process.

https://github.com/ansible/proposals/blob/master/proposals_process_proposal.md

If you have any further questions, please let us know by stopping by our devel mailing list, or our devel IRC channel:
- #ansible-devel on Freenode
- https://groups.google.com/forum/#!forum/ansible-devel

Thank you!

**Comment by dedekind** [comment]
Yeah, this would be a great feature to have!

**Comment by bcoca** [comment]
Added it, will appear in 2.2

**Comment by bobobox** [comment]
Awesome. Thank you.

**Comment by fuhbar** [comment]
@bcoca Great news, thank you! :)
